docs/packages/Parser.mdx

```diff
@@ -320,38 +321,49 @@ module.exports = {
 This option brings two main benefits over the older `project`:
 
 - Simpler configurations: most projects shouldn't need to explicitly configure `project` paths or create `tsconfig.eslint.json`s
-- Improved performance: this API is optimized on the TypeScript side for speed
-  - Initial versions of this option demonstrated performance changes in subsets of the typescript-eslint monorepo ranging from 11% slower to 70% faster
```

Comment:
We're still working on performance & filling in the gaps on edge cases. IMO we should stop bragging about it until we can be very, very certain it's better the vast majority of the time.